### PR TITLE
Emit TaskEvent messages for DevLoop, Build, and Deploy phases

### PR DESCRIPTION
**Pull request by MarlonGamez**
**Merge After**: #5641 
**Related**: #5368 #5422 
**Description**
Makes calls to eventV2 package to send out TaskEvent Protos. Also fixes what I believe to be an issue with the `SkaffoldRunner.devIteration` field not being updated after the first iteration

**User facing changes (remove if N/A)**
User can access v2 events endpoint and see TaskEvent messages being emitted. 

Example:
```json
{"result":{"timestamp":"2021-04-07T17:50:08.970854Z","taskEvent":{"id":"DevLoop-0","task":"DevLoop","iteration":0,"status":"InProgress","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:50:08.981571Z","taskEvent":{"id":"Build-0","task":"Build","iteration":0,"status":"InProgress","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:50:17.115852Z","taskEvent":{"id":"Build-0","task":"Build","iteration":0,"status":"Succeeded","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:50:17.990693Z","taskEvent":{"id":"Deploy-0","task":"Deploy","iteration":0,"status":"InProgress","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:50:22.981573Z","taskEvent":{"id":"Deploy-0","task":"Deploy","iteration":0,"status":"Succeeded","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:50:29.014458Z","taskEvent":{"id":"DevLoop-0","task":"DevLoop","iteration":0,"status":"Succeeded","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:53:33.602731Z","taskEvent":{"id":"DevLoop-1","task":"DevLoop","iteration":1,"status":"InProgress","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:53:33.604211Z","taskEvent":{"id":"Build-1","task":"Build","iteration":1,"status":"InProgress","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:53:48.021111Z","taskEvent":{"id":"Build-1","task":"Build","iteration":1,"status":"Succeeded","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:53:48.797537Z","taskEvent":{"id":"Deploy-1","task":"Deploy","iteration":1,"status":"InProgress","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:53:50.213166Z","taskEvent":{"id":"Deploy-1","task":"Deploy","iteration":1,"status":"Succeeded","actionableErr":null}}}
{"result":{"timestamp":"2021-04-07T17:53:54.512376Z","taskEvent":{"id":"DevLoop-1","task":"DevLoop","iteration":1,"status":"Succeeded","actionableErr":null}}}
```

**Follow-up Work (remove if N/A)**
Emit these messages for other phases of skaffold such as Test and StatusCheck

### CODE REVIEW

File: pkg/skaffold/runner/build_deploy.go [hunk 1]
```diff
@@ -87,6 +93,7 @@ func (r *SkaffoldRunner) Build(ctx context.Context, out io.Writer, artifacts []*
 	// Make sure all artifacts are redeployed. Not only those that were just built.
 	r.builds = build.MergeWithPreviousBuilds(bRes, r.builds)
 
+	eventV2.TaskSucceeded("Build", r.devIteration)
```
Comment (tejal29) [inline pkg/skaffold/runner/build_deploy.go:96]: use constants for `Build`, `Devloop` etc.

**Review by tejal29 — COMMENTED** [review]
This is great!
Can you add an example event stream ?

File: pkg/skaffold/runner/build_deploy.go (hunk 1 above)
Comment (MarlonGamez) [reply pkg/skaffold/runner/build_deploy.go]: done!

File: pkg/skaffold/server/server.go [hunk 2]
```diff
@@ -190,7 +190,7 @@ func newGRPCServer(preferredPort int, usedPorts *util.PortSet) (func() error, in
 }
 
 func newHTTPServer(preferredPort, proxyPort int, usedPorts *util.PortSet) (func() error, error) {
-	mux := runtime.NewServeMux(runtime.WithProtoErrorHandler(errorHandler))
+	mux := runtime.NewServeMux(runtime.WithProtoErrorHandler(errorHandler), runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{OrigName: true, EmitDefaults: true}))
```
Comment (tejal29) [inline pkg/skaffold/server/server.go:193]: nice!!!

**Review by tejal29 — APPROVED** [review]

**Pull request merged**